nova_vm/src/engine/bytecode/executable.rs

```diff
@@ -1313,12 +1313,30 @@ impl CompileEvaluation for ast::StaticMemberExpression<'_> {
             ctx.exe.add_instruction(Instruction::GetValue);
         }
 
+        ctx.exe.add_instruction(Instruction::LoadCopy);
+        if self.optional {
+            ctx.exe.add_instruction(Instruction::IsNullOrUndefined);
+        }
+
+        let jump_to_access = ctx
+            .exe
+            .add_instruction_with_jump_slot(Instruction::JumpIfNot);
+
+        ctx.exe.add_instruction(Instruction::Store);
         // 4. Return EvaluatePropertyAccessWithIdentifierKey(baseValue, IdentifierName, strict).
         let identifier = String::from_str(ctx.agent, self.property.name.as_str());
         ctx.exe.add_instruction_with_identifier(
             Instruction::EvaluatePropertyAccessWithIdentifierKey,
             identifier,
         );
+
+        let jump_to_end = ctx.exe.add_instruction_with_jump_slot(Instruction::Jump);
```

Comment:
**issue:** Here you can then conditionally produce the jump handling based on the Some(JumpIndex)/None from above. That way we don't need to add any jumps for when we do not have a conditional in the access.